### ISSUE TITLE
Editing doesn't work

### ISSUE THREAD

**Issue by aendra-rininsland** [akirk/friends#246]
I'm following my WordPress blog from my Mastodon user and AFAICT editing the post in WordPress doesn't push an edited version to the Fediverse — the copy of the post my Mastodon instance has cached doesn't have the changes I've made. 

I think to do this you need to generate a StatusEdit entity on every post edit. See https://docs.joinmastodon.org/entities/StatusEdit.

*Edit:* This might have more to do with `Automattic/wordpress-activitypub` than this repo; if so, please close this.

**Comment by akirk** [comment]
Thanks for reporting! Sending out this update is part of the ActivityPub plugin and [it supports sending out Update activities](https://github.com/Automattic/wordpress-activitypub/blob/0.17.0/includes/class-activitypub.php#L25).

The StatusEdit that you linked is from the Mastodon API which is used between a Mastodon client (like a mobile app) and a Mastodon server. Between Mastodon servers (and in this case your WordPress can be considered as such), the ActivityPub protocol is being used.

So if you don't see the updated text, maybe the ActivityPub Update activity wasn't received by your Mastodon server (as far as I know there is no retry cc @pfefferle).

Could you check your ActivityPub outbox to verify that it contains the updated content? If you're comfortable with the terminal you can do it like this: This is for how I would get my latest post, so replace the first URL

```
curl -LsH "Accept: application/activity+json" $author_url | jq -r .outbox
curl -LsH "Accept: application/activity+json" $previous_output | jq -r .first 
curl -LsH "Accept: application/activity+json" $previous_output | jq -r '.orderedItems[0].object.content'
```

So this is what it'd be for my own blog:

```
curl -LsH "Accept: application/activity+json" https://alex.kirk.at/author/alex | jq -r .outbox
curl -LsH "Accept: application/activity+json" https://alex.kirk.at/wp-json/activitypub/1.0/users/2/outbox | jq -r .first 
curl -LsH "Accept: application/activity+json" https://alex.kirk.at/wp-json/activitypub/1.0/users/2/outbox\?page\=1 | jq -r '.orderedItems[0].object.content'
```

**Comment by aendra-rininsland** [comment]
Got it, thanks! I also posted the same issue to the ActivityPub plugin and was told by the author it's a known issue:

https://github.com/Automattic/wordpress-activitypub/issues/409#issuecomment-1700954917

For the sake of completeness, I did the commands you requested and my updated content is there.

Closing because it's very clearly not a Friends issue at this point. Thanks for getting back to me, though! 💚
